### PR TITLE
Add tests on Python 3.12

### PR DESCRIPTION
**Pull request by dstansby**

### CODE REVIEW

File: .github/workflows/test_and_deploy.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
     strategy:
       matrix:
         platform: [ubuntu-latest, windows-latest, macos-latest]
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```
Comment (psobolewskiPhD) [inline .github/workflows/test_and_deploy.yml:28]:
```suggestion
        python-version: ["3.9", "3.10", "3.11", "3.12"]
```
napari no longer supports 3.8 (and soon 3.9), so I suggest dropping it here to keep the same number of CI runs

File: setup.cfg [hunk 2]
```diff
@@ -18,6 +18,7 @@ classifiers =
 	Programming Language :: Python :: 3.9
```
Comment (psobolewskiPhD) [inline setup.cfg:18]: Can't suggest, but I suggest deleting line 17, Python 3.8

File: tox.ini [hunk 3]
```diff
@@ -1,13 +1,14 @@
 # For more information about tox, see https://tox.readthedocs.io/en/latest/
 [tox]
-envlist = py{38,39,310,311}-{linux,macos,windows}
+envlist = py{38,39,310,311,312}-{linux,macos,windows}
```
Comment (psobolewskiPhD) [inline tox.ini:3]:
```suggestion
envlist = py{39,310,311,312}-{linux,macos,windows}
```

File: tox.ini [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
 
 [gh-actions]
 python =
     3.8: py38
```
Comment (psobolewskiPhD) [inline tox.ini:7]:
```suggestion
```

**Comment by dstansby** [comment]
I'd rather keep this PR simple, and remove Python 3.8 in another PR.

**Comment by jburel** [comment]
Thanks, I am currently removing Python 3.8 from various repository and updating the submission to pypi (see #120)
I will open a PR now removing Python 3.8 support (EOL 07/10/2024)

**Comment by dstansby** [comment]
🎉 thanks1

**Pull request merged**